common/src/main/java/org/conscrypt/HpkeContextRecipient.java

```diff
@@ -16,85 +16,91 @@
-     * <p>
-     * Note: This API keeps track of its state. It maintains an internal HPKE context. As a result,
-     * to decrypt multiple messages that were encrypted using the same context, one must call this
-     * method for each message in the same order as they were encrypted.
+     * Returns an uninitialised HpkeContextRecipient from a specific {@link Provider}
      *
-     * @param ciphertext contains the encrypted plaintext
-     * @param aad        optional associated data
-     * @return plaintext
-     * @throws IllegalStateException if an issue is encountered while decrypting (an issue could
-     *         occur
-     *                               most likely if the keys configured are not valid)
+     * @param suite the HPKE suite to use.  @see {@link HpkeSuite} for details.
+     * @param provider the Provider to use
+     * @return an uninitialised HpkeContextRecipient for the requested suite
+     * @throws NoSuchAlgorithmException if no implementation could be found
+     * @throws NoSuchProviderException if providerName is null or no such Provider exists
      */
-    public byte[] open(byte[] ciphertext, byte[] aad) {
-        Preconditions.checkNotNull(ciphertext, "ciphertext");
-        try {
-            return NativeCrypto.EVP_HPKE_CTX_open(ctx, ciphertext, aad);
-        } catch (Exception e) {
-            throw new IllegalStateException(
-                    "Error while decrypting with the keys provided during setup recipient", e);
-        }
+    public static HpkeContextRecipient getInstance(String suite, Provider provider)
+        throws NoSuchAlgorithmException, NoSuchProviderException {
+        return new HpkeContextRecipient(findSpi(suite, provider));
     }
 
     /**
-     * Hybrid Public Key Encryption (HPKE) secret export.
+     * Initialises this HpkeContextRecipient.
      *
-     * @param length          expected output length
-     * @param exporterContext optional exporter context
-     * @return exported value
-     * @throws IllegalArgumentException if the length is not valid based on the KDF spec
+     * @param mode HPKE mode to use, currently only HpkeContext.MODE_BASE is supported.
+     * @param encapsulated encapsulated ephemeral key from an HpkeContextSender
+     * @param privateKey private key of the recipient
+     * @param info application-supplied information, may be null or empty
+     * @throws InvalidKeyException if privateKey is null or an unsupported key format
+     * @throws UnsupportedOperationException if mode is not a supported HPKE mode
+     * @throws IllegalStateException if this HpkeContextRecipient has already been initialised
      */
-    public byte[] export(int length, byte[] exporterContext) {
-        hpkeSuite.getKdf().validateExportLength(length);
-        return NativeCrypto.EVP_HPKE_CTX_export(ctx, exporterContext, length);
+    public void init(int mode, byte[] encapsulated, PrivateKey privateKey, byte[] info)
```

Comment:
An issue I see here is with the "mode" parameter. If mode is BASE_MODE, the rest of the parameters are ok. If later, we expand and support AUTH, PSK, or AUTH_PSK, we don't have all the parameters. My suggestion would be to have 4 different "init" methods to initialize the `HpkeContextRecipient` and you won't need to pass the mode, it can be inferred. Just adding 1 "init" method should be ideal for now as the other modes are not supported yet. 

BASE
enc, skR, info

AUTH
enc, skR, info, pkS

PSK
enc, skR, info, psk, pid

AUTH & PSK
enc, skR, info, psk, pid, pkS